### DISCUSSION TITLE
Use in Vision OS?

### DISCUSSION THREAD

**Discussion by dxu** [jrouwe/JoltPhysics#976] [Q&A]
Hi, I'm a fan of the engine, coming from Godot! I was wondering if you had advice on how to use this in a VisionOS project? Currently, I'm trying to compile this as a static library that I could use for an interactive app. My understanding is that I'd have to include Objective C bridging headers in order to use it with swift. 

I'm not an expert with C++ or Swift, but am trying to learn, so any direction you could point me in, or prior work, would be greatly appreciated!

**Comment by jrouwe** [comment]
I'm sorry, my last objective C experience is probably more than 10 years old so I can't help you there. I would expect this to work the same for all C++ libraries though, so you should be able to get some help by googling around.

**Comment by meshula** [comment]
@dxu Very recent Xcodes have got native swift C++ interop. I'd recommend studying the documentation (https://www.swift.org/documentation/cxx-interop/)  and having a go, it's new to all of us, but people are getting good results wrapping various projects so far (e.g. https://github.com/wabiverse/SwiftUSD)
